### ISSUE TITLE
WebVTT, Text and SRT support

### ISSUE THREAD

**Issue by nodomain** [fourTheorem/podwhisperer#6]
Are there plans yet to support other output formats like WebVTT, plain text or SRT?

Still digging through the solution and thinking about adding a converter but I am not sure about the correct approach.

**Comment by eoinsha** [comment]
Yes, this is definitely something we have discussed and would help improve things like YouTube subtitles.
AWS Transcribe supports generation of WebVTT and SRT subtitles already. Of course, we would prefer to use Whisper segments instead of Transcribe for the subtitle text. The challenge here is that the timing granularity of Whisper output is not as fine as with Transcribe, so you end up with longish segments for each timestamp. This may not be desirable for many subtitle uses.

It may be possible to improve the merging algorithm to match the Transcribe timings to the Whisper output, but that seems not so trivial.

Perhaps the simplest thing initially is to just generate VTT or SRT from the Whisper output. This could be done in a separate Lambda function using the merged transcript output (after the [`Process Transcripts`](https://github.com/fourTheorem/podwhisperer/blob/5c0e00783bec280953142574a9f9803e229f594c/transcript-orchestration/statemachine/transcription-step-function.asl.json#L320) state).
It could use the JSON object located at `processedTranscriptKey` as its input.

If you are interested in contributing this feature, that would be very welcome, @nodomain. We are happy to review and support of course.

**Comment by nodomain** [comment]
Looking into it already :)
